lib/fast-path.js

```diff
@@ -183,13 +183,21 @@ FPp.map = function map(callback/*, name1, name2, ... */) {
 // Inspired by require("ast-types").NodePath.prototype.needsParens, but
 // more efficient because we're iterating backwards through a stack.
 FPp.needsParens = function(assumeExpressionContext) {
+  var name = this.getName();
+  var node = this.getNode();
+
+  // This needs to come before `if (!parent) { return false }` because
+  // an object destructuring assignment requires parens for
+  // correctness even when it's the topmost expression.
+  if (node.type === "AssignmentExpression" && node.left.type === 'ObjectPattern') {
+    return true;
+  }
+
   var parent = this.getParentNode();
   if (!parent) {
     return false;
   }
 
-  var name = this.getName();
```

Comment:
Might want to keep the `name` line down here, since root/parentless nodes don't have names?